# Titanic Data Analysis

## Importing packages and the data set

In [4]:
# Basic package imports
import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
import plotly.express as px
from plotly.subplots import make_subplots
import plotly.io as pio
pio.renderers.defaule = 'colab'

In [5]:
import kagglehub
# Download latest version
path = kagglehub.dataset_download("yasserh/titanic-dataset")
print("Path to dataset files:", path)

Path to dataset files: C:\Users\josee\.cache\kagglehub\datasets\yasserh\titanic-dataset\versions\1


In [8]:
# Copy the path to the data set
file = '/Users/josee/.cache/kagglehub/datasets/yasserh/titanic-dataset/versions/1/Titanic-Dataset.csv'
df = pd. read_csv(file)
df

PassengerId  Survived  Pclass  \
0              1         0       3   
1              2         1       1   
2              3         1       3   
3              4         1       1   
4              5         0       3   
..           ...       ...     ...   
886          887         0       2   
887          888         1       1   
888          889         0       3   
889          890         1       1   
890          891         0       3   

                                                  Name     Sex   Age  SibSp  \
0                              Braund, Mr. Owen Harris    male  22.0      1   
1    Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                               Heikkinen, Miss. Laina  female  26.0      0   
3         Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                             Allen, Mr. William Henry    male  35.0      0   
..                                                 ...     ...   ...    ...   
886                              Montvila, Rev. Juozas    male  27.0      0   
887                       Graham, Miss. Margaret Edith  female  19.0      0   
888           Johnston, Miss. Catherine Helen "Carrie"  female   NaN      1   
889                              Behr, Mr. Karl Howell    male  26.0      0   
890                                Dooley, Mr. Patrick    male  32.0      0   

     Parch            Ticket     Fare Cabin Embarked  
0        0         A/5 21171   7.2500   NaN        S  
1        0          PC 17599  71.2833   C85        C  
2        0  STON/O2. 3101282   7.9250   NaN        S  
3        0            113803  53.1000  C123        S  
4        0            373450   8.0500   NaN        S  
..     ...               ...      ...   ...      ...  
886      0            211536  13.0000   NaN        S  
887      0            112053  30.0000   B42        S  
888      2        W./C. 6607  23.4500   NaN        S  
889      0            111369  30.0000  C148        C  
890      0            370376   7.7500   NaN        Q  

[891 rows x 12 columns]

## 1. How many variables and observations? Which are Numerical/Categorical?

In [9]:
df.shape

(891, 12)

There are 891 observations and 12 variables.

In [11]:
df.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

**Variable Notes**
- Categorical:
    - Survived: Survived (1) or died (0)
    - Pclass: Passenger’s class (1st, 2nd, or 3rd)
    - Name: Passenger’s name
    - Sex: Passenger’s sex
    - Ticket: Ticket number
    - Fare: Fare paid for ticket
    - Cabin: Cabin number
    - Embarked: Where the passenger got on the ship (C — Cherbourg, S — Southampton, Q = Queenstown)

- Numerical:
    - PassengerId: Unique ID of the passenger
    - Age: Passenger’s age
    - SibSp: Number of siblings/spouses aboard the Titanic
    - Parch: Number of parents/children aboard the Titanic

## 2. Do any of the columns have NaNs in them? What do NaNs mean?

In [19]:
df.isna().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

'Age' has 177 NaNs, 'Cabin' has 687 NaNs, and 'Embarked' has 2 NaNs. This means an observation is lacking data for a certain variable. It is somewhat concerning that 'Cabin' has so many NaNs. Since it has fewer observations, results regarding 'Cabin' should be approached with caution.

## 3. How many passengers survived?